src/Databases/DatabasesCommon.cpp

```diff
@@ -247,6 +259,29 @@ StoragePtr DatabaseWithOwnTablesBase::detachTableUnlocked(const String & table_n
 
     return res;
 }
+StoragePtr DatabaseWithOwnTablesBase::detachTableCaseInsensitiveUnlocked(const String & table_name)
```

Comment:
Have you thought about unifying `detachTableUnlocked` and `detachTableCaseInsensitiveUnlocked`? I think the only difference would be using `operator==` or `boost::iequals`, while it would make the code less bugprone as it would avoid duplicating logic.